apollo-router/src/spec/schema.rs

```diff
@@ -308,23 +317,31 @@ impl Schema {
                         .definitions()
                         .filter_map(|definition| {
                             if let $ast_ty(definition) = definition {
-                                let instance = <$ty>::from(definition);
-                                Some((instance.name.clone(), instance))
+                                match <$ty>::try_from(definition) {
+                                    Ok(instance) => Some(Ok((instance.name.clone(), instance))),
+                                    Err(e) => Some(Err(e)),
+                                }
                             } else {
                                 None
                             }
                         })
-                        .collect::<HashMap<String, $ty>>();
+                        // todo: impl from
+                        .collect::<Result<HashMap<String, $ty>, _>>()
+                        .map_err(|e| SchemaError::Api(e.to_string()))?;
 
                     document
                         .definitions()
                         .filter_map(|definition| {
                             if let $ast_extension_ty(extension) = definition {
-                                Some(<$ty>::from(extension))
+                                Some(<$ty>::try_from(extension))
                             } else {
                                 None
                             }
                         })
+                        // todo: impl from
```

Comment:
Should we delete it ?

Reply:
Yeah, we ll deal with it once we refactor this bit of code away!